src/materialized/lib.rs

```diff
@@ -55,7 +55,17 @@ pub const VERSION: &str = env!("CARGO_PKG_VERSION");
 pub const BUILD_SHA: &str = run_command_str!(
     "sh",
     "-c",
-    r#"[ -n "$MZ_DEV_BUILD_SHA" ] && echo "$MZ_DEV_BUILD_SHA" || git rev-parse --verify HEAD"#
+    r#"if [ -n "$MZ_DEV_BUILD_SHA" ]; then
+        echo "$MZ_DEV_BUILD_SHA"
+    else
+        # Unfortunately we need to suppress error messages from `git`, as
+        # run_command_str will display no error message at all if we print
+        # more than one line of output to stderr.
+        git rev-parse --verify HEAD 2>/dev/null || {
+            echo "error: unable to determine Git SHA; either build from working Git clone, or specify SHA manually by setting MZ_DEV_BUILD_SHA environment variable" >&2;
```

Comment:
Link to https://materialize.io/docs/install/#build-from-source ?

Reply:
good idea, done